modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceRecordSearchTest.java

```diff
@@ -149,6 +149,88 @@ public void testExactPhrase() throws Exception {
 		assertSearch("Bloggs", 2);
 	}
 
+	@Test
+	public void testMultipleLanguageEntries() throws Exception {
+		List<KeyValuePair> keyValuePairs = new ArrayList<>();
+
+		keyValuePairs.add(new KeyValuePair("name", "keyword"));
+		keyValuePairs.add(new KeyValuePair("description", "keyword"));
+
+		DDMForm ddmForm = createDDMForm(
+			keyValuePairs, LocaleUtil.US, LocaleUtil.BRAZIL);
+
+		for (DDMFormField ddmFormField : ddmForm.getDDMFormFields()) {
+			LocalizedValue label = ddmFormField.getLabel();
+
+			label.addString(LocaleUtil.BRAZIL, label.getString(LocaleUtil.US));
+
+			ddmFormField.setLabel(label);
+		}
+
+		DDMFormInstance ddmFormInstance =
+			DDMFormInstanceTestUtil.addDDMFormInstance(
+				ddmForm, _group, _user.getUserId());
+
+		_searchContext = getSearchContext(_group, _user, ddmFormInstance);
+
+		DDMFormValues ddmFormValues1 =
+			DDMFormValuesTestUtil.createDDMFormValues(
+				ddmForm,
+				DDMFormValuesTestUtil.createAvailableLocales(LocaleUtil.US),
+				LocaleUtil.US);
+
+		DDMFormFieldValue nameDDMFormFieldValue =
+			createLocalizedDDMFormFieldValue(
+				"name",
+				HashMapBuilder.put(
+					LocaleUtil.US, "United States"
+				).build());
+
+		ddmFormValues1.addDDMFormFieldValue(nameDDMFormFieldValue);
+
+		DDMFormFieldValue descriptionDDMFormFieldValue =
+			createLocalizedDDMFormFieldValue(
+				"description",
+				HashMapBuilder.put(
+					LocaleUtil.US, "Solo"
+				).build());
+
+		ddmFormValues1.addDDMFormFieldValue(descriptionDDMFormFieldValue);
+
+		DDMFormInstanceRecordTestUtil.addDDMFormInstanceRecord(
+			ddmFormInstance, ddmFormValues1, _group, _user.getUserId());
+
+		DDMFormValues ddmFormValues2 =
+			DDMFormValuesTestUtil.createDDMFormValues(
+				ddmForm,
+				DDMFormValuesTestUtil.createAvailableLocales(LocaleUtil.BRAZIL),
+				LocaleUtil.US);
+
+		nameDDMFormFieldValue = createLocalizedDDMFormFieldValue(
+			"name",
+			HashMapBuilder.put(
+				LocaleUtil.BRAZIL, "Brasil"
+			).build());
+
+		ddmFormValues2.addDDMFormFieldValue(nameDDMFormFieldValue);
+
+		descriptionDDMFormFieldValue = createLocalizedDDMFormFieldValue(
+			"description",
+			HashMapBuilder.put(
+				LocaleUtil.BRAZIL, "Solo"
+			).build());
+
+		ddmFormValues2.addDDMFormFieldValue(descriptionDDMFormFieldValue);
+
+		DDMFormInstanceRecordTestUtil.addDDMFormInstanceRecord(
+			ddmFormInstance, ddmFormValues2, _group, _user.getUserId());
```

Comment:
Can you reuse the methods from this test class?

See the test methods in this class and you can see how we can reuse the existing methods to create your scenario.

Reply:
This:
```
List<KeyValuePair> keyValuePairs = new ArrayList<>();

keyValuePairs.add(new KeyValuePair("name", "keyword"));
keyValuePairs.add(new KeyValuePair("description", "keyword"));

DDMForm ddmForm = createDDMForm(
	keyValuePairs, LocaleUtil.US, LocaleUtil.BRAZIL);

for (DDMFormField ddmFormField : ddmForm.getDDMFormFields()) {
	LocalizedValue label = ddmFormField.getLabel();

	label.addString(LocaleUtil.BRAZIL, label.getString(LocaleUtil.US));

	ddmFormField.setLabel(label);
}

DDMFormInstance ddmFormInstance =
	DDMFormInstanceTestUtil.addDDMFormInstance(
		ddmForm, _group, _user.getUserId());
```

Could be replaced by:
```
List<KeyValuePair> keyValuePairs = new ArrayList<>();

keyValuePairs.add(new KeyValuePair("name", "keyword"));
keyValuePairs.add(new KeyValuePair("description", "keyword"));

DDMFormInstance ddmFormInstance =
	DDMFormInstanceTestUtil.addDDMFormInstance(
		createDDMForm(keyValuePairs, LocaleUtil.US, LocaleUtil.BRAZIL), _group,
		_user.getUserId());
```

Reply:
You could use or adapt the method `addDDMFormInstanceRecord()` instead of:

```
DDMFormValues ddmFormValues1 =
	DDMFormValuesTestUtil.createDDMFormValues(
		ddmForm,
		DDMFormValuesTestUtil.createAvailableLocales(LocaleUtil.US),
		LocaleUtil.US);

DDMFormFieldValue nameDDMFormFieldValue =
	createLocalizedDDMFormFieldValue(
		"name",
		HashMapBuilder.put(
			LocaleUtil.US, "United States"
		).build());

ddmFormValues1.addDDMFormFieldValue(nameDDMFormFieldValue);

DDMFormFieldValue descriptionDDMFormFieldValue =
	createLocalizedDDMFormFieldValue(
		"description",
		HashMapBuilder.put(
			LocaleUtil.US, "Solo"
		).build());

ddmFormValues1.addDDMFormFieldValue(descriptionDDMFormFieldValue);

DDMFormInstanceRecordTestUtil.addDDMFormInstanceRecord(
	ddmFormInstance, ddmFormValues1, _group, _user.getUserId());

DDMFormValues ddmFormValues2 =
	DDMFormValuesTestUtil.createDDMFormValues(
		ddmForm,
		DDMFormValuesTestUtil.createAvailableLocales(LocaleUtil.BRAZIL),
		LocaleUtil.US);

nameDDMFormFieldValue = createLocalizedDDMFormFieldValue(
	"name",
	HashMapBuilder.put(
		LocaleUtil.BRAZIL, "Brasil"
	).build());

ddmFormValues2.addDDMFormFieldValue(nameDDMFormFieldValue);

descriptionDDMFormFieldValue = createLocalizedDDMFormFieldValue(
	"description",
	HashMapBuilder.put(
		LocaleUtil.BRAZIL, "Solo"
	).build());

ddmFormValues2.addDDMFormFieldValue(descriptionDDMFormFieldValue);

DDMFormInstanceRecordTestUtil.addDDMFormInstanceRecord(
	ddmFormInstance, ddmFormValues2, _group, _user.getUserId());
```